### PR TITLE
transfer-contract: Validate tx hash

### LINKED ISSUE #402: Calculate the transaction hash inside of the contract
**Describe what you want implemented** 
Provide a function in the transfer contract to reconstruct the transaction hash

**Describe "Why" this is needed**
We need to build the circuit public inputs inside of the contract. It means the transaction hash will have to be calculated there.

**Describe alternatives you've considered**
Using a constant as transaction hash, but that is highly insecure.

**Additional context**
https://github.com/dusk-network/rusk/blob/c5f0170b033691639fb9c3e40facf7508fd7706f/contracts/transfer/src/wasm/internal.rs#L190-L193

### PR DESCRIPTION
**Pull request by vlopes11**
Resolves #402

### CODE REVIEW

File: contracts/transfer/src/transfer.rs [hunk 1]
```diff
@@ -96,6 +97,37 @@ impl TransferContract {
...
+        hasher.update(fee.to_bytes());
+
+        if let Some(c) = crossover {
+            hasher.update(c.to_bytes());
+        };
+
+        if let Some((cid, txdata)) = call {
+            hasher.update(cid.as_bytes());
+
+            // TODO review if as_bytes of `Transaction` is the same as canon
```
Comment (ureeves) [inline contracts/transfer/src/transfer.rs:123]: `Canon` is no longer used to do this in `dusk-wallet-core`. As far as the wallet is concerned the transaction is just a `Vec<u8>`.

**Review by ureeves — APPROVED** [review]
LGTM, even with the nit

**Review by ZER0 — APPROVED** [review]
LGTM

File: contracts/transfer/src/transfer.rs (hunk 1 above)
Comment (vlopes11) [reply contracts/transfer/src/transfer.rs]: Gotcha, removed the comment

**Pull request merged**